ArduCopter/mode_guided.cpp

```diff
@@ -286,47 +321,86 @@ bool ModeGuided::set_destination(const Location& dest_loc, bool use_yaw, float y
         pos_control_start();
     }
 
-    if (!wp_nav->set_wp_destination_loc(dest_loc)) {
-        // failure to set destination can only be because of missing terrain data
-        AP::logger().Write_Error(LogErrorSubsystem::NAVIGATION, LogErrorCode::FAILED_TO_SET_DESTINATION);
-        // failure is propagated to GCS with NAK
-        return false;
-    }
-
     // set yaw state
     set_yaw_state(use_yaw, yaw_cd, use_yaw_rate, yaw_rate_cds, relative_yaw);
 
+    // set position target and zero velocity and acceleration
+    if (!dest_loc.get_vector_from_origin_NEU(guided_pos_target_cm)) {
+        guided_pos_target_cm = pos_control->get_pos_target_cm();
+    }
+    guided_vel_target_cms.zero();
+    guided_accel_target_cmss.zero();
+    vel_update_time_ms = millis();
```

Comment:
I don't think we should be setting this vel_update_time_ms timer.  This is for detecting the case where the user or companion computer hasn't sent in a velocity target (and maybe it is also being re-used to detect if the user hasn't sent in an acceleration target).  In either case, I don't think we should be resetting it in the case where we've received a position target and the vehicle is happy to proceed all the way to the target (even if it is far away) without any further updates from the user.

Reply:
replaced both with update_time_ms

Reply:
Yep, removed